### DISCUSSION TITLE
any plans of supporting SQLite

### DISCUSSION THREAD

**Discussion by anantshri** [pixelfed/pixelfed#5118] [Support]
I would love to selfhost pixelfed instance for just myself i.e. a single user instance. It would be awesome if there could be a trimmed down version of software with lesser system resource requirements as the use-case is of single user only. 

One of the major hurdles become the database support can that be moved to sqlite3 this reduces a big overhead. similarly can redis be reduced from the single user setup.

**Comment by sandervankasteel** [comment]
It's definitely possible to do that. Pixelfed uses Laravel and Laravel has native support for SQLite. See [https://laravel.com/docs/10.x/database#sqlite-configuration](https://laravel.com/docs/10.x/database#sqlite-configuration) for more information 

But in short, all you need to do is replace your `DATABASE_` configuration values with the following configuration;

```
DB_CONNECTION=sqlite
DB_DATABASE=/absolute/path/to/database.sqlite
``` 

You might also want to set `DB_FOREIGN_KEYS=true`, but play around with it.
